plugins/frontend/backstage-plugin-argo-cd/src/api/index.ts

```diff
@@ -99,11 +115,38 @@ export class ArgoCDApiClient implements ArgoCDApi {
     );
   }
 
-  async getAppDetails(options: { url: string; appName: string }) {
-    const proxyUrl = await this.getProxyUrl();
+  async getAppDetails(options: {
+    url: string;
+    appName: string;
+    instance?: string;
+  }) {
+    const proxyUrl = await this.getBaseUrl();
+    if (this.searchInstances === true) {
+      return this.fetchDecode(
+        `${proxyUrl}/argoInstance/${options.instance}/applications/${options.appName}`,
+        argoCDAppDetails,
+      );
+    }
     return this.fetchDecode(
       `${proxyUrl}${options.url}/applications/${options.appName}`,
       argoCDAppDetails,
     );
   }
+
+  async argoServiceLocator(options: { appName: string }) {
+    const proxyUrl = await this.getBaseUrl();
```

Comment:
Small thing, but could you rename this to `serviceLocatorUrl` or `ArgoBackendUrl` or something. It will always call the backend portion of this plugin in this case right?

Reply:
addressed